backend/src/services/user.service.ts

```diff
@@ -0,0 +1,44 @@
+import UserModel, { IUser } from '../models/User';
+import bcrypt from 'bcrypt';
+import User from '@interfaces/User';
+export const createUser = async (user: User): Promise<IUser> => {
+  try {
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(user.password, salt);
+    const newUser = new UserModel({
+      userName: user.userName,
+      email: user.email,
+      password: hashedPassword,
+    });
+    const savedUser = await newUser.save();
+    return savedUser;
+  } catch (error) {
+    throw new Error(`Error creating user: ${error}`);
+  }
+};
+
+export const getUserByEmailOrUserName = async (identifier: string) => {
```

Comment:
Use better names to better reflect their purpose without implying statefulness.

```suggestion
export const getUser = async (identifier: string) => {
```